docs/book/v2/quick-start.md

```diff
@@ -0,0 +1,144 @@
+example, so working knowledge of the other sections of the manual is suggested.
+
+## Example Application
+
+Assume, for a moment, the following application code. It already assumes that
+dependencies are injected, and so becomes a good candiate for a DiC.
+
+```php
+namespace MyLibrary
+{
+    class DbAdapter
+    {
+        protected $username = null;
+        protected $password = null;
+
+        public function __construct($username, $password)
+        {
+            $this->username = $username;
+            $this->password = $password;
+        }
+    }
+}
+
+namespace MyMovieApp
+{
+    class MovieFinder
+    {
+        protected $dbAdapter = null;
+
+        public function __construct(\MyLibrary\DbAdapter $dbAdapter)
+        {
+            $this->dbAdapter = $dbAdapter;
+        }
+    }
+
+    class MovieLister
+    {
+        protected $movieFinder = null;
+
+        public function __construct(MovieFinder $movieFinder)
+        {
+            $this->movieFinder = $movieFinder;
+        }
+    }
+}
+```
+
+With the above code, you find yourself writing the following to wire and utilize
+it:
+
+```php
+// $config object is assumed
+
+$dbAdapter = new MyLibrary\DbAdapter($config->username, $config->password);
+$movieFinder = new MyMovieApp\MovieFinder($dbAdapter);
+$movieLister = new MyMovieApp\MovieLister($movieFinder);
+foreach ($movieLister as $movie) {
+    // iterate and display $movie
+}
+```
+
+If you are doing this above wiring in each controller or view that wants to list
+movies, not only can this become repetitive and boring to write, but also
+unmaintainable if you want to swap out one of these dependencies on a wholesale
+scale.
+
+Since this example of code already practices good dependency injection using
+constructor injection, it is a great candidate for using laminas-di.
+
+## Using a laminas-di Container
+
+The following demonstrates how to wire the above into a laminas-di container:
+
+```php
+// Inside a bootstrap somewhere
+$di = new Laminas\Di\Di();
+$di->instanceManager()->setParameters('MyLibrary\DbAdapter', [
+    'username' => $config->username,
+    'password' => $config->password,
+]);
+
+// Elsewhere:
+$movieLister = $di->get('MyMovieApp\MovieLister');
+foreach ($movieLister as $movie) {
+    // iterate and display $movie
+}
+```
+
+In the above example, we are obtaining a *default instance* of `Laminas\Di\Di`. By
+'default', we mean that `Laminas\\Di\\Di` is constructed with a `DefinitionList`
+seeded with a `RuntimeDefinition` (which uses PHP's Reflection API) and an empty
+instance manager and no configuration:
+
+```php
+public function __construct(
+    DefinitionList $definitions = null,
+    InstanceManager $instanceManager = null,
+    Configuration $config = null
+) {
+    $this->definitions = ($definitions) ?: new DefinitionList(new Definition\RuntimeDefinition());
+    $this->instanceManager = ($instanceManager) ?: new InstanceManager();
```

Comment:
I think the full class names with namespaces should be here.

Reply:
This is (incorrectly) mirroring to the constructor code of `Laminas\Di\Di` (v2): https://github.com/laminas/laminas-di/blob/2.6.1/src/Di.php#L113

Reply:
I changed it to an example code snippet